src/parsers/trades/liquid/index.ts

```diff
@@ -0,0 +1,88 @@
+import { getCSVData } from '../../';
+import { EXCHANGES, IImport, IPartialTrade, ITrade } from '../../../types';
+import { createDateAsUTC, createID } from '../../utils';
+
+interface ILiquid {
+    exchange: string;
+    currency_type: string;
+    direction: string;
+    transaction_id: string;
+    transaction_type: string;
+    gross_amount: string;
+    currency: string;
+    execution_id: string;
+    generated_for_type: string;
+    generated_for_id: string;
+    transaction_hash: string;
+    from_address: string;
+    to_address: string;
+    state: string;
+    created_at_jst: string;
+    updated_at_jst: string;
+    created_at_utc: string;
+    updated_at: string;
+    notes: string;
+    used_id: string;
+    account_id: string;
+}
+
+export default async function processData(importDetails: IImport): Promise<ITrade[]> {
+    const data: ILiquid[] = await getCSVData(importDetails.data) as ILiquid[];
+    const internalFormat: ITrade[] = [];
+    if (data.length < 1) {
+        return internalFormat;
+    }
+    let splitTrade = data[0];
```

Comment:
Just to confirm this is because again 1 trade = 2 line items?

What about if we did a for loop here(instead of for of) and then just incremented 2 each time?

Reply:
I think the new solution is much better. This liquid file has some very strange and inconsistent looking data as well, worse than Revolut.